src/discord.coffee

```diff
@@ -58,22 +95,28 @@ class DiscordBot extends Adapter
      message: (message) =>
         # ignore messages from myself
         return if message.author.id == @client.user.id
-        user                      = @robot.brain.userForId message.author.id
-        user.room                 = message.channel.id
-        user.name                 = message.author.username
-        user.discriminator        = message.author.discriminator
-        user.id                   = message.author.id
-
-        @rooms[message.channel.id]?= message.channel
 
-        text = message.cleanContent
-
-        if (message?.channel instanceof Discord.DMChannel)
-          text = "#{@robot.name}: #{text}" if not text.match new RegExp( "^@?#{@robot.name}" )
+        user = @_map_user message.author, message.channel.id
+        text = @_format_incoming_message(message)
 
         @robot.logger.debug text
         @receive new TextMessage( user, text, message.id )
 
+     message_reaction: (reaction_type, message, user) => 
+        # ignore reactions from myself
+        return if user.id == @client.user.id
+
+        reactor = @_map_user user, message.message.channel.id
+        author = @_map_user message.message.author, message.message.channel.id
+        text = @_format_incoming_message message.message
+
+        text_message = new TextMessage(reactor, text, message.message.id)
+        reaction_string = message._emoji.name
+        if message._emoji.id?
+          reaction_string += ":#{message._emoji.id}"
+        @receive new ReactionMessage(reaction_type, reactor, reaction_string,
```

Comment:
maybe keep `reaction_string` consistent between constructor and usage here so it's easier to track where in the code we are, as well as some of the others

Reply:
I'm not sure what you mean by this.

Reaction string differs between custom and standard emotes.  Standard emotes come through with a null id which we don't append.

Reply:
I'm referring to variable naming